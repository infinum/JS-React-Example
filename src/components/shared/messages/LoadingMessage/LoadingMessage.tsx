import React, { FC } from 'react';
import { Center, CenterProps, Image, Text } from '@chakra-ui/react';

export interface ILoadingMessage extends CenterProps {
	message?: string;
}

export const LoadingMessage = ({ message = 'Loading...', ...centerProps }: ILoadingMessage) => (
	<Center h="100%" {...centerProps}>
		<Image w="240px" alt="presentation" src="/images/logo-infinum.png" />
		<Text ml={10} color="red" fontSize="3xl">
			{message}
		</Text>
	</Center>
);
