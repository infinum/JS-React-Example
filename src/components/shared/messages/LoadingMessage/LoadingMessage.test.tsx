import { axe } from 'jest-axe';

import { render } from 'test-utils';
import { LoadingMessage } from './LoadingMessage';

describe('LoadingMessage', () => {
	it('should matches snapshot', () => {
		const { asFragment } = render(<LoadingMessage />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should be accessible', async () => {
		const { container } = render(<LoadingMessage />);

		const results = await axe(container);

		expect(results).toHaveNoViolations();
	});
});
