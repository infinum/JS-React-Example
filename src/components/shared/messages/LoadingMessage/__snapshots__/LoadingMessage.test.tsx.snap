// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingMessage should matches snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 100%;
}

.emotion-1 {
  width: 240px;
}

.emotion-2 {
  margin-left: var(--chakra-space-10);
  color: red;
  font-size: var(--chakra-fontSizes-3xl);
}

<div
    class="emotion-0"
  >
    <img
      class="chakra-image emotion-1"
      role="presentation"
      src="/images/logo-infinum.png"
    />
    <p
      class="chakra-text emotion-2"
    >
      Loading...
    </p>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
